api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -77,6 +77,21 @@ public interface Scan<ThisT, T extends ScanTask, G extends ScanTaskGroup<T>> {
    */
   ThisT includeColumnStats();
 
+  /**
+   * Create a new scan from this that loads the column stats for the specific columns with each data
+   * file.
+   *
+   * <p>Column stats include: value count, null value count, lower bounds, and upper bounds.
+   *
+   * @param requestedColumns column names for which to keep the stats. If <code>null</code> then all
```

Comment:
I am not sure how I feel about supporting `null` here. Let me think.
Also, won't the current implementation throw an exception if we pass `null`?

Reply:
I think we should not support null here as a valid value independently of what we will do in `ContentFile`. I would drop the doc.

Reply:
I would guess that the situation is the similar with the other `@Nullable` `TableScanContext` attributes, like:
- snapshotId
- selectedColumns
- projectedSchema
- fromSnapshotId
- toSnapshotId
- branch

We have an undefined default behaviour which could be archived with not setting the values, or setting them to `null`. The only difference here is that we define this default behaviour. For consistency's sake we can remove the comment, but the behaviour will remain the same.

Do I miss something?

Thanks for the detailed review!